### PR TITLE
PLAT-494: test deduplication of method calls using VFindCallRequest

### PR DESCRIPTION
**Pull request by ruz**

### CODE REVIEW

File: ledger-core/virtual/integration/deduplication_test.go [hunk 1]
```diff
@@ -318,3 +321,525 @@ func TestDeduplication_SecondCallOfMethodAfterExecution(t *testing.T) {
...
+		{
+			name: "no pendings, findCall message, found call, has result",
+
+			expectFindRequestMessage: true,
+			findRequestStatus:        payload.FoundCall,
+			findRequestHasResult:     true,
+
+			expectResultMessage: true,
+		},
+
```
Comment (aalekseev-ru) [inline ledger-core/virtual/integration/deduplication_test.go:387]: there must be test "no pendings, findCall message, found call, no result",

Comment (ruz) [reply ledger-core/virtual/integration/deduplication_test.go:387]: I don't think so. I believe that it's situation that shouldn't happen in release 0. If previous executor knows about request then executor either produced result or request is in pending executions, but we don't have pendings, so it has result. In future releases such case may happen once we start "forgetting" results.

Comment (ruz) [reply ledger-core/virtual/integration/deduplication_test.go:387]: I would rather see panic Impossible if we got "found without result", pendings table is complete and request is not in the table, but I didn't write test for that.

File: ledger-core/virtual/integration/deduplication_test.go [hunk 2]
```diff
@@ -318,3 +321,525 @@ func TestDeduplication_SecondCallOfMethodAfterExecution(t *testing.T) {
...
+func (s *deduplicateMethodUsingPrevVETest) getClass() reference.Global {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+
+	return s.class
+}
+
+func (s *deduplicateMethodUsingPrevVETest) getOutgoingRef() reference.Global {
+	callee := s.getObject() // TODO: FIXME: should be caller
```
Comment (aalekseev-ru) [inline ledger-core/virtual/integration/deduplication_test.go:661]: is there jira issue for this?

Comment (ruz) [reply ledger-core/virtual/integration/deduplication_test.go:661]: Nope.

File: ledger-core/virtual/integration/deduplication_test.go (hunk 1 above)
Comment (aalekseev-ru) [reply ledger-core/virtual/integration/deduplication_test.go:387]: I can make panic in my code and add a test for that case

File: ledger-core/virtual/integration/deduplication_test.go [hunk 3]
```diff
@@ -318,3 +321,525 @@ func TestDeduplication_SecondCallOfMethodAfterExecution(t *testing.T) {
...
+func (s *deduplicateMethodUsingPrevVETest) waitMessagePublications(
+	ctx context.Context,
+	t *testing.T,
+	expected int,
+) {
+	t.Helper()
+	if !s.server.PublisherMock.WaitCount(expected, 10*time.Second) {
+		require.Fail(t, "timeout waiting for messages on publisher")
+	}
+	require.Equal(t, expected, s.server.PublisherMock.GetCount())
```
Comment (ElSacramento) [inline ledger-core/virtual/integration/deduplication_test.go:831]: You can use it
```
	require.Equal(t, 4, typedChecker.VCallRequest.Count())
	require.Equal(t, 2, typedChecker.VCallResult.Count())
```

Comment (ruz) [reply ledger-core/virtual/integration/deduplication_test.go:833]: done

File: ledger-core/virtual/integration/deduplication_test.go (hunk 2 above)
Comment (ruz) [reply ledger-core/virtual/integration/deduplication_test.go]: created

File: ledger-core/virtual/integration/deduplication_test.go (hunk 3 above)
Comment (ElSacramento) [reply ledger-core/virtual/integration/deduplication_test.go:833]: function `waitMessagePublications` can be removed now, am I right?

Comment (ruz) [reply ledger-core/virtual/integration/deduplication_test.go]: yep, thanks, done

**Pull request merged**